src/tiramisu_core.cpp

```diff
@@ -6145,6 +6166,153 @@ void computation::split(int L0, int sizeX)
+
+    int n_dims = isl_map_dim(this->get_schedule(), isl_dim_out);
+    std::vector<isl_id *> dimensions;
+    std::vector<std::string> dimensions_str;
+    std::string map = "{";
+
+    // -----------------------------------------------------------------
+    // Preparing a map to split the duplicate computation.
+    // -----------------------------------------------------------------
+
+    map = map + this->get_name() + "[";
+
+    for (int i = 0; i < n_dims; i++)
+    {
+        if (i == 0)
+        {
+            std::string dim_str = generate_new_variable_name();
+            dimensions_str.push_back(dim_str);
+            map = map + dim_str;
+        }
+        else
+        {
+            std::string dim_str = generate_new_variable_name();
+            dimensions_str.push_back(dim_str);
+            map = map + dim_str;
+
+            if (i == inDim0)
+            {
+                inDim0_str = dim_str;
+            }
+        }
+
+        if (i != n_dims - 1)
+        {
+            map = map + ",";
+        }
+    }
+
+    map = map + "] -> " + this->get_name() + "[";
+
+    for (int i = 0; i < n_dims; i++)
+    {
+        if (i == 0)
+        {
+            map = map + dimensions_str[i];
+            dimensions.push_back(isl_id_alloc(
+                                     this->get_ctx(),
+                                     dimensions_str[i].c_str(),
+                                     NULL));
+        }
+        else if (i != inDim0)
+        {
+            map = map + dimensions_str[i];
+            dimensions.push_back(isl_id_alloc(
+                                     this->get_ctx(),
+                                     dimensions_str[i].c_str(),
+                                     NULL));
+        }
+        else
+        {
+            map = map + outDim0_str + ", " + static_dim_str + ", " + outDim1_str;
+            isl_id *id0 = isl_id_alloc(this->get_ctx(),
+                                       outDim0_str.c_str(), NULL);
+            isl_id *id2 = isl_id_alloc(this->get_ctx(),
+                                       static_dim_str.c_str(), NULL);
+            isl_id *id1 = isl_id_alloc(this->get_ctx(),
+                                       outDim1_str.c_str(), NULL);
+            dimensions.push_back(id0);
+            dimensions.push_back(id2);
+            dimensions.push_back(id1);
+        }
+
+        if (i != n_dims - 1)
+        {
+            map = map + ",";
+        }
+    }
+
+    /**
+     * The main idea is to avoid a case when the iterations dont start from 0.
+     * let assume it start from 1, in that case after splitting of \p i into \p i1 & \p i2 with factor 4, the result would be:
+     *  in first iteration of \p i1 would would compute \p i2 = [1,2,3,4] then [5,6,7,8]
+     *  Whereas in the legacy version in first iteration of \p i1 would would compute \p i2 = [1,2,3]  then [4,5,6,7]
+     * 
+    */
+    inDim0_str =  "("+inDim0_str +" - "+lower_bound+")";
+
```

Comment:
This code is a copy/paste from split, the main and core difference is in this line that changes everything.
inDim0_str =  "("+inDim0_str +" - "+lower_bound+")"; makes all iterations with the same number of points which solves our issues we had.